common/build.gradle.kts

```diff
@@ -15,6 +15,9 @@ dependencies {
     compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
     implementation(libs.bundles.semconv)
 
+    implementation(libs.contribSpanStacktrace)
```

Comment:
IMO this should not be a implementation (and therefore transitive) depenency of our common project. This way, we pull in the extension into other extensions, which we don't want.

See also my other comment about not replacing our local copy of `MutableSpan`.

Instead, this dependency needs to be compile-only because we still need to support unwrapping the `MutableSpans` if that extension happens to be on the classpath. You'll have to revisit the span unwrapping logic in `SpanValue`.

I'd suggest adding a `static final Map<Class<?>, Function<Span,Span>> UNWRAPPER_REGISTRY` and populating it dynamically based on whether the `MutableSpan` shipped with the span-stacktrace extension is found on the classpath at runtime or not.

Reply:
Or, easier solution: Just have a `List<Function<Span,Span>> UNWRAPPERS` populated based on the classpath.
The functions return the unwrapped span (if they know the type and can unwrap) or return the provided span as-is.

The unwrapping logic would then do the following:
```
public Span unwrap(Span toUnwrap) {
    Span previous = null;
    Span current = toUnwrap;
    while (previous != current) {
        previous = current;
        for (Function<Span,Span> unwrapper: UNWRAPPERS) {
             current = unwrapper.apply(current);
        }
    }
    return current;
}
```

Reply:
I think you are right here, we should keep the current MutableSpan and remove it once we have a cleaner solution with the API.